### PR TITLE
Fix controls

### PR DESCRIPTION
**Pull request by icemac**
* Fix set value to something falsy for controls besides radio buttons.
* Match parts of the display title. 

Restore behavior to the old one of mechanize.

### CODE REVIEW

File: src/zope/testbrowser/browser.py [hunk 1]
```diff
@@ -847,7 +847,9 @@ def displayValue(self, value):
 
         values = []
         for key, titles in self._getOptions():
-            if any(t in value for t in titles):
+            if any(v in t
+                   for t in titles
+                   for v in value):
```
Comment (mgedmin) [inline src/zope/testbrowser/browser.py]: What happens if I assing a string to `displayValue`?  I don't think character-by-character iteration is right here.

Comment (icemac) [inline src/zope/testbrowser/browser.py]: In f949a8c I added handling for strings.

**Comment by icemac** [comment]
@mgedmin Here are the tests. :)

File: src/zope/testbrowser/tests/test_browser.py [hunk 2]
```diff
@@ -82,6 +83,45 @@ def __call__(self, environ, start_response):
...
+  <body>
+    <form>
+      <select name="sel1">
+        <option value="op">Turn</option>
+        <option value="alt">Alternative</option>
+      </select>
+    </form>
+  </body>
+</html>
+''')
```
Comment (mgedmin) [inline src/zope/testbrowser/tests/test_browser.py]: Personally I find tests more readable when they use [textwrap.dedent](https://docs.python.org/2/library/textwrap.html#textwrap.dedent) to have the entire test body indented more than the `def` line.  Also that way my Vim folding doesn't break ;)

**Pull request merged**